### ISSUE TITLE
Pasting multiline text in tiny opens $EDITOR but then the message is sent even if not saved when quitting in $EDITOR

### ISSUE THREAD

**Issue by Kabouik** [osa1/tiny#345]
Pretty much what the title says:

![ss-2021-08-03_110707](https://user-images.githubusercontent.com/7107523/127989558-389101f9-8fe4-465f-96f0-1230b2647b4a.png)

**Comment by osa1** [comment]
How this works is, when you do `C-x` or paste multi-line text we run `$EDITOR` on a temporary file. The file initially has the text you pasted. If your editor exists with 0 then we use the contents of the temporary file as the message. So if you want to abort pasting in `$EDITOR`, then you either have to make your editor exit with some code other than 0, or make sure that the file is empty (just delete the contents and save).

In principle we could check for modification date to see whether the file was saved or not, but that would have other problems. For example, I often want to send the pasted text directly without modification. In those cases tiny runs `$EDITOR` and I immediately close it, without saving it again (reminder that the file initially has the pasted text). If we check for whether the file was updated or not, then that would not send the message.

**Comment by Kabouik** [comment]
I see, thanks @osa1.

Then maybe a simple change would be to clarify that in the comment block, with something like `Beware that any uncommented text below will be sent to tiny even if the buffer is not saved when quitting the text editor. Make sure you delete all text and save the buffer to cancel sending the message.`?

If I understand correctly what you described, the following will still send text:

1. quit without saving when there are uncommented lines at the time `$EDITOR` is started (after pasting text in tiny)
2. quit and save with uncommented lines (of course)
3. delete uncommented lines automatically added when pasting multi-line in tiny, and quit `$EDITOR` without saving, because then file won't be changed and it will contain uncommented text

The following will not send text:
4. `C-x` then quit without saving
5. `C-x` then quit with saving before adding uncommented text
6. Paste multi-line text in tiny, then delete uncommented lines in  `$EDITOR` and save

I believe 1 and 4 are not trivial, as well as 3 to a lesser extent (if you don't save the file, you should expect that whatever you did before will have no effect) because users probably assume that the buffer is a scratch buffer, and should be cancelled if not saved. So a more detailed description (yet as short as possible, not like my message here :<) would help and possibly avoid regrettable problems like pasting a wrong clipboard by mistake and sharing your "hunter2" IRC password or your email address with everyone.

**Comment by osa1** [comment]
You got it right @Kabouik . Looking at the current message

```
# You pasted a multi-line message. When you close the editor final version of
# this file will be sent (ignoring these lines). Delete contents to abort the
# paste.
```

I think this already conveys the right message, in the "final version of this version" part. At least to me, it's clear that final version is the version in your last save. (we should update "You pasted a multi-line message" part as that's not the case when you do `C-x`)

I'm still happy to update it if anyone has a better (ideally concise) phrasing.

**Comment by Kabouik** [comment]
You are right, this already convers the right message, it's me who overlooked it. I would maybe just suggest something like:

```
# You pasted multi-line text or opened the editor manually. When you quit the editor,
# the final version of this file will be sent (ignoring these lines), whether you save it or
# not. To abort and send no message, make sure you delete the content below.
```

But I don't know if this clarifies this much. Another option would be to \*not\* send the message to IRC when quitting the editor, and just hadd the text content in the Tiny input field. This would require an extra `Return` to send, but it's also a safety and I think it's the expected behavior when invoking `$EDITOR` in any TUI program, including email clients: usually it just allows complex editing but sends the text content to the TUI program's builtin text field instead of confirming the message.
